### PR TITLE
fix: resolve unresolved Copilot comments from PR #55

### PR DESCRIPTION
**Pull request by irvinesunday**
## Summary
Addresses all Copilot review comments that were unresolved when PR #55 (fix/locality-display-names) was auto-merged into develop.

## Session / Phase
Session: Maintenance — fix
Phase: Phase 1 — Citizen Mobile

## Agent C Verdict
Verdict: N/A
Approved to merge: Yes

## Coverage
Line coverage: PENDING
Gate status: PENDING

### Already addressed at HEAD by post-merge edits (verified)
- `NominatimGeocodingService.cs` — `System.Web.HttpUtility` replaced with `Uri.EscapeDataString`; `HttpResponseMessage` wrapped in `using` (comments #3044783556, #3044783599).
- `FollowedLocalitiesRequestDto.cs` — `[StringLength(160)]` added to `FollowedLocalityItemDto.DisplayLabel` so over-length input is rejected at the model-binding boundary instead of triggering a `DbUpdateException` (comment #3044783696).
- `LocalitiesController.cs` — `TrimLabel` comment now matches the implementation ("first 2 segments") (comment #3044783734).

### Fixed in this PR
- **`src/Hali.Application/Notifications/FollowService.cs`** (comments #3044783640 and #3044783674):
  - `SetFollowedAsync` dedupe now prefers the first entry carrying a non-empty `DisplayLabel` so a later null/blank entry can no longer silently clobber a meaningful label.
  - `GetFollowedWithDetailsAsync` now drops follows whose locality cannot be resolved from the Signals DB instead of returning a `FollowedLocalityDto` with an empty `WardName` (the API contract treats `wardName` as required).
- **`tests/Hali.Tests.Unit/Notifications/FollowServiceTests.cs`** (comment #3044783756):
  - Adds 4 unit tests for the new behavior:
    1. `GetFollowedWithDetails_ReturnsDisplayLabel_WhenPresent`
    2. `GetFollowedWithDetails_ReturnsNullDisplayLabel_WhenNotProvided`
    3. `GetFollowedWithDetails_DropsFollows_WhenLocalityLookupMissing`
    4. `SetFollowed_DedupePrefersNonNullDisplayLabel`

## How to test
1. `dotnet test tests/Hali.Tests.Unit/Hali.Tests.Unit.csproj` — 172/172 passing locally.
2. Verify the previously flagged patterns no longer exist at HEAD:
   - `git show HEAD:src/Hali.Infrastructure/Signals/NominatimGeocodingService.cs | grep -n "HttpUtility\|System.Web"` — no matches.
   - `git show HEAD:src/Hali.Application/Notifications/FollowService.cs | grep -n "g.First()"` — only the dedupe fallback line, after the non-null preference filter.
   - `git show HEAD:src/Hali.Application/Notifications/FollowService.cs | grep -n "string.Empty"` — no matches in `GetFollowedWithDetailsAsync`.

## Checklist
- [ ] All 6 CI jobs are green
- [ ] Coverage gate ≥ 95%
- [ ] Agent C validation report reviewed
- [x] No hardcoded secrets or API keys
- [x] No .env files committed
- [x] EF Core migrations are reversible
- [x] Outbox events written in same transaction as state changes
- [x] No features outside MVP scope introduced

### CODE REVIEW

File: tests/Hali.Tests.Unit/Notifications/FollowServiceTests.cs [hunk 1]
```diff
@@ -104,4 +119,97 @@ public async Task SetFollowed_ReplacesExistingFollows()
...
+        await svc.SetFollowedAsync(accountId, new[]
+        {
+            new FollowEntry(localityId, null),
+            new FollowEntry(localityId, "South B"),
+        });
+
+        var follows = await svc.GetFollowedAsync(accountId);
+        var follow = Assert.Single(follows);
+        Assert.Equal("South B", follow.DisplayLabel);
+    }
```
[the remark this answers is not in the document]
Comment (irvinesunday) [reply tests/Hali.Tests.Unit/Notifications/FollowServiceTests.cs]: Fixed in commit e691d68.

Problem: SetFollowedAsync uses IsNullOrWhiteSpace to detect a missing DisplayLabel, but the existing dedupe test only covered the null→non-null case, leaving "", "   ", and "\t" unprotected against regressions.
Fix applied: Added a [Theory] SetFollowed_DedupeTreatsBlankDisplayLabelAsMissing with InlineData(""), InlineData("   "), and InlineData("\t") that asserts the labelled entry wins when the same localityId arrives first with a blank/whitespace label.
Files changed: tests/Hali.Tests.Unit/Notifications/FollowServiceTests.cs
Verified: `dotnet test --filter FullyQualifiedName~FollowServiceTests` → 11 passed, 0 failed (3 new theory cases included).

[the remark this answers is not in the document]
Comment (irvinesunday) [reply tests/Hali.Tests.Unit/Notifications/FollowServiceTests.cs]: Confirmed resolved. Verified at HEAD: FollowServiceTests.SetFollowed_DedupeTreatsBlankDisplayLabelAsMissing is a [Theory] with InlineData "", "   ", and "\t" (lines 216-238). It asserts that a blank/whitespace DisplayLabel does not clobber a real labelled entry in the same SetFollowedAsync call. No further changes needed.

**Comment by irvinesunday** [comment]
PR cleanup pass complete for #57. Branch synced with develop. Copilot comments resolved: 1 (confirmed already addressed by existing [Theory] test). CI fixes applied: none required — all checks green after develop sync. Merge method: --merge (admin). Merged at: 2026-04-08T07:42:32Z.
